src/popupmenu.c

```diff
@@ -574,6 +574,306 @@ pum_user_attr_combine(int idx, int type, int attr)
     return user_attr[type] > 0 ? hl_combine_attr(attr, user_attr[type]) : attr;
 }
 
+#ifdef FEAT_RIGHTLEFT
+/*
+ * Display RTL text with proper attributes in the popup menu.
+ * Returns the adjusted column position after drawing.
+ */
+    static int
+pum_display_rtl_text(
+	int	row,
+	int     col,
+	char_u  *st,
+	int     attr,
+	int     *attrs,
+	int     width,
+	int     width_limit,
+	int     totwidth,
+	int     next_isempty)
+{
+    char_u  *rt;
+    int     cells;
+    int     over_cell = 0;
+    int     truncated = FALSE;
+    int     pad = next_isempty ? 0 : 2;
+    int     remaining;
+    int	    truncrl = '<';
+
+    if (curwin->w_fill_chars.truncrl != NUL)
+	truncrl = curwin->w_fill_chars.truncrl;
+
+    if (st == NULL)
+	return col;
+
```

Comment:
Move this check before the `truncrl` assignment? This makes `pum_display_rtl_text()` and `pum_display_ltr_text()` more consistent.